### DISCUSSION TITLE
Harbor LDAP setting in harbor.yml

### DISCUSSION THREAD

**Discussion by mephmanx** [goharbor/harbor#16927] [Ideas]
All,

Is there any way to add LDAP setting to the harbor.yml so that everything is configured on deployment instead of having to log in to the UI and update manually?  I would like to setup filly automated deployments of harbor instances.  If its not available currently....would it be something that would fit with the project?  i.e. is there any reason why it wouldent work or wasnt implemented already?

**Comment by stonezdj** [comment]
https://github.com/goharbor/website/blob/6d7c3dffc46aae30a75fa08dd74466418e63145c/docs/install-config/configure-user-settings-cli.md

**Reply by mephmanx** [reply]
Thats a very interesting doc...many thanks for pointing that out!  I think the override part will work nicely for me...just need to test it out.  Thank you again!!

**Reply by mephmanx** [reply]
I wasnt able to get it to work setting it as and env variable on the vm running harbor but I am reading it more closely and it seems to say "set as a variable on the core container".  So, should it be set on in the docker container that is running the core?

CONTAINER ID   IMAGE                                COMMAND                  CREATED         STATUS                            PORTS                                                                            NAMES
053209dbba85   goharbor/harbor-jobservice:v2.5.1    "/harbor/entrypoint.…"   7 seconds ago   Up 5 seconds (health: starting)                                                                                    harbor-jobservice
d95e210eef44   goharbor/nginx-photon:v2.5.1         "nginx -g 'daemon of…"   7 seconds ago   Up 5 seconds (health: starting)   0.0.0.0:80->8080/tcp, :::80->8080/tcp, 0.0.0.0:443->8443/tcp, :::443->8443/tcp   nginx
0bc1a71e73bb   goharbor/harbor-core:v2.5.1          "/harbor/entrypoint.…"   7 seconds ago   Up 6 seconds (health: starting)                                                                                    harbor-core
1904bbf24c35   goharbor/redis-photon:v2.5.1         "redis-server /etc/r…"   8 seconds ago   Up 7 seconds (health: starting)                                                                                    redis
81a65c098bab   goharbor/harbor-db:v2.5.1            "/docker-entrypoint.…"   8 seconds ago   Up 7 seconds (health: starting)                                                                                    harbor-db
75aaef62f808   goharbor/harbor-portal:v2.5.1        "nginx -g 'daemon of…"   8 seconds ago   Up 7 seconds (health: starting)                                                                                    harbor-portal
a160406d2ea8   goharbor/harbor-registryctl:v2.5.1   "/home/harbor/start.…"   8 seconds ago   Up 7 seconds (health: starting)                                                                                    registryctl
ca7b81a93cfc   goharbor/registry-photon:v2.5.1      "/home/harbor/entryp…"   8 seconds ago   Up 7 seconds (health: starting)                                                                                    registry
a8182784d8dc   goharbor/harbor-log:v2.5.1           "/bin/sh -c /usr/loc…"   9 seconds ago   Up 8 seconds (health: starting)   127.0.0.1:1514->10514/tcp                                                        harbor-log

For example, in the above setup, I need to set it in this container?

0bc1a71e73bb

**Comment by mephmanx** [comment]
This worked perfectly once I added it to the environment of the core container.  Thank you so much for this!
